### ISSUE TITLE
Logo for RosettaSciIO

### ISSUE THREAD

**Issue by jlaehne** [hyperspy/rosettasciio#5]
Was just playing with an idea for a logo for RosettaSciIO, based on some hieroglyphs (see https://de.wikipedia.org/wiki/%C3%84gyptische_Hieroglyphen). Symbols for R S TT I are used:

![image](https://user-images.githubusercontent.com/7076057/180884396-5540f454-cfaa-4d40-9629-8c90c96a055a.png)

Dark and bright versions, with either grayshades in the text or pure b/w.

**Comment by jat255** [comment]
Wikimedia has a nice 3D model of the real rosetta stone: https://commons.wikimedia.org/wiki/File:Rosetta_Stone.stl Perhaps we could work that into a logo idea somehow? It would get across the idea of "translation"

**Comment by jlaehne** [comment]
Yes, thought about at least playing with the contour of the stone, which I find clearer than the 3D object (a bit complicated for a logo):

![image](https://user-images.githubusercontent.com/7076057/180989256-a77ad804-2711-4c4c-9821-c8941f0d66dc.png)

The 3D object would be under [CC-BY-NC-SA](https://www.myminifactory.com/object-licensing), while the contour was derived from a public domain work: https://commons.wikimedia.org/wiki/File:Rosetta_Stone_BW.jpeg

**Comment by ericpre** [comment]
Thanks @jlaehne, For a square logo, personally, I find line 4, column 1 and 3 in https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1195326090 quite nice. :) 

It is possible to make one that work on both dark and light background?

**Comment by jat255** [comment]
My concern is that the "hieroglyphs" don't make it obvious what this library does or is even called. I like the concept of using the stone, but maybe we make the name more obvious? 

Some quick mockups:

(I actually quite like this)
![image](https://user-images.githubusercontent.com/1278301/184962027-0cbacda3-79c6-4f59-8e88-fb9e831f2e6c.png)

(I don't love this next one... but as more of a logo idea)
![image](https://user-images.githubusercontent.com/1278301/184962300-1e11b661-a27f-4e35-b92f-90b10c1eff71.png)

(maybe we make the acronym 'RSIO'?)
![image](https://user-images.githubusercontent.com/1278301/184962527-993d7252-3eb4-42be-8d2b-220b04f49b8b.png)

Here's an SVG with those mockups in case anyone else wants to play:

![rosettasciio_logo_mockups](https://user-images.githubusercontent.com/1278301/184962748-93fcbbdf-3213-4454-9909-98551e17efc6.svg)

**Comment by CSSFrancis** [comment]
@jat255 Those look pretty good.  I think the top one is the best as far as maybe something like a banner.  As far as a logo goes I think that we could probably just have the Rossetta stone with IO on it.

Maybe we think about adding some science "hieroglyphs" so that all parts of the name are part of the logo, something like a Bohr model of atom, a bandgap or a beaker etc.

**Comment by jlaehne** [comment]
Lets narrow down the choices for the Logo, please everyone vote using thumbs up/down on the following questions:

**Comment by jlaehne** [comment]
Black/white and white/black (thumbs up)
versus background-agnostic gray scale (thumbs down)

**Comment by jlaehne** [comment]
greek letters instead of a/e (thumbs up)
versus stick with latin letters (thumbs down)

**Comment by jlaehne** [comment]
Font from png in https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1217620339 - jlaehne (thumbs up)
versus font from png in https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1217054931 - jat255 (thumbs down)

**Comment by ericpre** [comment]
@jordiferrero, interested to do some of your magic? :)

**Comment by jordiferrero** [comment]
Thanks @ericpre for flagging this :-) I really love the suggestions already proposed. In fact, I was envisioning the exact same idea... the stone with the "IO" as writing.
I personally love simple flat design. So I took @jat255 [simple design](https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1217054931) and simply added the "Python" snake in the S of RosettaSci... as a hint of it being a Python package. Attached the .svg files in case you want to add it.

_Note: I did not know what font you used, so I vectorised the font, that's why it's not prefect..._

<img width="596" alt="logo@2x" src="https://user-images.githubusercontent.com/9082687/189213947-ab97a131-2715-49ab-9f11-b767f8ea2ba6.png">
<img width="199" alt="icon@2x" src="https://user-images.githubusercontent.com/9082687/189213950-00f65c97-c228-4e58-884c-545f561014b7.png">

**Comment by jordiferrero** [comment]
> Thanks @ericpre for flagging this :-) I really love the suggestions already proposed. In fact, I was envisioning the exact same idea... the stone with the "IO" as writing. I personally love simple flat design. So I took @jat255 [simple design](https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1217054931) and simply added the "Python" snake in the S of RosettaSci... as a hint of it being a Python package. Attached the .svg files in case you want to add it.
> 
> _Note: I did not know what font you used, so I vectorised the font, that's why it's not prefect..._
> 
> <img alt="logo@2x" width="596" src="https://user-images.githubusercontent.com/9082687/189213947-ab97a131-2715-49ab-9f11-b767f8ea2ba6.png">
> 
> <img alt="icon@2x" width="199" src="https://user-images.githubusercontent.com/9082687/189213950-00f65c97-c228-4e58-884c-545f561014b7.png">

![logo](https://user-images.githubusercontent.com/9082687/189214562-8a7a18b1-5b49-4b37-94b9-1825189c633f.svg)

**Comment by jat255** [comment]
> I did not know what font you used

It's Lato: https://www.latofonts.com/

**Comment by ericpre** [comment]
Thank you @jordiferrero. Time for a poll of two questions then!

**Comment by ericpre** [comment]
normal S (thumbs up)
S with snake (thumbs down)

**Comment by ericpre** [comment]
IO only inside the stone (thumbs up)
more letters in the stone (RSIO, RScIO) (thumbs down)

**Comment by hakonanes** [comment]
don't know if it's a problem really, but might there be people reading "IO" as "ten"?

**Comment by ericpre** [comment]
Yes good point, it may be worth considering `I/O` or lower case `io`. Can someone have a go at making different versions, please? :)

**Comment by jlaehne** [comment]
> Yes good point, it may be worth considering `I/O` or lower case `io`. Can someone have a go at making different versions, please? :)

There should be fonts, like the one of these comments, which are Sans Serif, but still have short horizontal lines on the upper case I, which would make it clear to be IO and not 10.

**Comment by jlaehne** [comment]
Though, I do like Lento, @jordiferrero would it be possible to add the horizontal lines on the I (like in Tahoma, Verdana, Noto, Segoe UI) on your design otherwise keeping the font?

**Comment by jordiferrero** [comment]
> Though, I do like Lento, @jordiferrero would it be possible to add the horizontal lines on the I (like in Tahoma, Verdana, Noto, Segoe UI) on your design otherwise keeping the font?

![image](https://user-images.githubusercontent.com/9082687/189635849-f7416279-b058-43b9-97f2-ca3c30ae8657.png)

Some iterations on your ideas

**Comment by francisco-dlp** [comment]
+1 for the original (f)—I just find its minimalism more appealing.

Would it be possible (and good) to keep the space between characters uniform to emphasize that it is a single word? Currently IO seems separated from the rest.

**Comment by francisco-dlp** [comment]
Talking about minimalism, what about remove "osetta" from the logo i.e. RSciIO with IO inside the stone?

**Comment by jat255** [comment]
> what about remove "osetta" from the logo

As a logo for use in presentations, documentation, etc., I'd rather have the full name in the logo. I was partial to a smaller "RS/IO" in the stone as a "square" format logo (I also think the import should be `rsio`, but that ship has probably sailed :smile: ), but I understand if that's not a shared opinion.

**Comment by jlaehne** [comment]
> (I also think the import should be `rsio`, but that ship has probably sailed)

v0.1 has not been released yet, so I would not say that it has sailed, but we should nail it now.

The double 'i' in the import is prone to typos, so that would be an argument for the even shorter version.

**Comment by ericpre** [comment]
I agree that it is worth considering `rsciio` versus `rsio`, as it is typo prone.

When asking around about `IO` versus `io`, people who are not familiar with what `IO` means read it as 10... so even if option f is the one I prefer, I think the best is option d.

**Comment by francisco-dlp** [comment]
The problem with `io` in the logo is that it changes the "official" style of the name from RosettaSciIO to RosettaSciio. Personally, I very much prefer the former.

**Comment by ericpre** [comment]
Yes, I know, but the fact that everyone (4-5 people) that I have asked read it as 10, it did put me off from option f!
As the `io` is in the stone and is distinctive from the rest of the name, I think this is fine and a good compromise.

**Comment by jat255** [comment]
Based off the other discussion, I agree that lower-case is probably better for the logo. I'm partial to either *b* or *d*. One issue with *d* to me is that the two "i" characters right next to each other are slightly different size, which is a bit jarring. *b* utilizes a different font, so it's not so obvious or off-putting being a different size.

**Comment by ericpre** [comment]
> One issue with _d_ to me is that the two "i" characters right next to each other are slightly different size, which is a bit jarring

Yes indeed, can it be fixed? :)

**Comment by francisco-dlp** [comment]
A remark regarding lowercase `io`: that's not an acronym but a [moth](https://en.wikipedia.org/wiki/Automeris_io). Here is its chenille, that, now that we are at it, could replace the Python in the logo ;)  ![io chenille](https://upload.wikimedia.org/wikipedia/commons/8/87/Caterpillar_in_Gemini_Springs%2C_Florida_-_Flickr_-_Andrea_Westmoreland.jpg)?

**Comment by jat255** [comment]
> that's not an acronym but a [moth](https://en.wikipedia.org/wiki/Automeris_io)

It's also a moon! https://en.wikipedia.org/wiki/Io_(moon)

**Comment by jordiferrero** [comment]
@francisco-dlp Here is the change in logo f. The spacing between the lower case i and the higher case I is x3 that of the space between the c and the i in "Sci". I have reduced it to what it was before. The reason the spacing is not x1 is that the black stone background acts as a "character" and you need a bit of extra spacing otherwise the I cannot be read.

![image](https://user-images.githubusercontent.com/9082687/190897907-548962c2-cdc6-4b7e-99a8-77565e622917.png)

@jat255 @ericpre Here are versions b-d with improved spacing and sizing.
![image](https://user-images.githubusercontent.com/9082687/190898619-bc003fd1-5b05-4d1a-908c-1dd797d6b92d.png)

I personally prefer the lower case ones (b first, then d). I also asked some colleagues and they read 10 instead of IO...


Finally, here is a mock for the shorter versions (I don't particularly like any of them more than the previous designs)...
![image](https://user-images.githubusercontent.com/9082687/190898986-32da7207-195b-4211-bd5b-974b311b1ac9.png)

**Comment by jlaehne** [comment]
I find it weird to have two different lower case `i` in version b) and still see the general problem with having lowercase in the name of these cenral letters.
To resolve the issue with the 10, I still tend towards e) - potentially with less thick 'serifs' or even slightly slanted ones:

![image](https://user-images.githubusercontent.com/7076057/190923980-c75add9e-b3a7-4c28-a224-ba44b5b27403.png)
(of course the snake needs to be added still)

**Comment by jat255** [comment]
I quite like @jordiferrero's new version (b). Using a different font further accentuates that "io" is an important part of the package, and using a monospaced serif font goes with the "programming" theme. 

I do not care for the serif capitalized "IO" (sorry @jlaehne!)

**Comment by francisco-dlp** [comment]
This is becoming the most contentious issue in HyperSpy's history...

-1 for lowercase `io`: while it solves the confusion with 10, it makes it harder to understand what `io` means. Also, I am with @jlaehne, I find it less aesthetically pleasing. Therefore, if we must solve the confusion with 10, just like @jlaehne,  +1 for version e).

Regarding short vs long version, there are multiple options:

| | Name | Library | Logo |
| - | -------- | --------- | ------- |
| a | RosettaSciIO | rosettasciio | RosettaSciIO |
| b / b2 | RosettaSciIO | rsciio / rsio | RosettaSciIO |
| c / c2 | RosettaSciIO | rsciio / rsio | RSciIO / RSIO |
| d / d2 | RSciIO / RSCI | rsciio / rsio | RSciIO / RSIO |

+1 for a) and c)

**Comment by francisco-dlp** [comment]
Another for the IO in the logo is adding a dot of top of the I, like this **İ**.

**Comment by ericpre** [comment]
I agree that variant e) is a good comprise:
- it is capitalised - in line with the name
- it can't be misread as 10

I was initially put off by using different fonts but as @jat255 mentioned [above](https://github.com/hyperspy/rosettasciio/issues/5#issuecomment-1250387628), it is actually useful to make it more distinctive.

Even if this is a bit tedious, it is worth taking the time to get it as good as we can from the start. In any case, this can be improved later, as this wouldn't be set in stone - pun intended! ;)

**Comment by jlaehne** [comment]
I found it OK to modulate Lato, because in principle I did like Lato and there are other Sans Serif fonts with such an I, e.g. "Segoe UI", "Sen", "Amble", "Antic Sans":

![image](https://user-images.githubusercontent.com/7076057/191069553-dd5d9506-1724-4552-abda-b37ec0b48517.png)

But I could also live with the Code fonts you propose.

**Comment by francisco-dlp** [comment]
I find that Antic Sans goes well with the ancient roots of the project.

**Comment by jordiferrero** [comment]
> I found it OK to modulate Lato, because in principle I did like Lato and there are other Sans Serif fonts with such an I, e.g. "Segoe UI", "Sen", "Amble", "Antic Sans":
> 
> ![image](https://user-images.githubusercontent.com/7076057/191069553-dd5d9506-1724-4552-abda-b37ec0b48517.png)
> 
> But I could also live with the Code fonts you propose.

Perfect. It seems we are now converging. I would suggest using the thicker fonts because (i) it works better for small logos + miniaturizations and (ii) it allows for the addition of the snake (which I am a big fan of). I like Amber more of these 3...

**Comment by jlaehne** [comment]
If you want thicker fonts, I even found a few more ;-) To make the choice even harder ( I did not like the squared dot on the i for Amble):

![image](https://user-images.githubusercontent.com/7076057/191259821-f6f1d451-0da8-4b6b-9deb-af43489458bf.png)

**Comment by francisco-dlp** [comment]
I like Noto's I, +1 for it.

**Comment by jat255** [comment]
I prefer the straight serifs, but happy to put it to a vote. I'm good with eitehr Radio Canada or Harmattan. Here it is with some tweaked sizing on the `IO` part (`I` is exact same height as the `R` and `S`):

![image](https://user-images.githubusercontent.com/1278301/191557340-ca3ee1e2-501c-4899-87f8-fa4a9fe0dbf8.png)

I guess @jlaehne and I must have used slightly different methods to vectorize the stone outline, but I prefer the slightly softer edges in my version:

This the SVG, which doesn't display properly for me in Chrome (probably since it's an obscure font -- possibly something to be aware of):
![184962748-93fcbbdf-3213-4454-9909-98551e17efc6](https://user-images.githubusercontent.com/1278301/191557367-d9f3f70e-adf7-489c-9f1e-cbadf2747d3a.svg)

**Comment by jlaehne** [comment]
I can live with both Noto or Radio Canada ;-)

> I guess @jlaehne and I must have used slightly different methods to vectorize the stone outline, but I prefer the slightly softer edges in my version:

As long as it is vectorized in the final version, I'm happy ;-) I used Inkscape, but even that has multiple options.

> This the SVG, which doesn't display properly for me in Chrome (probably since it's an obscure font -- possibly something to be aware of):

The final version should probably have the text converted to paths to avoid any font issues!

**Comment by ericpre** [comment]
Before we start suffering from logo fatigue syndrome, I would suggest that we down select features with some polls.
Once we have a good idea of what features are the most popular, it will be a matter of refining the alignment, height, width, etc. 😊

I will try to select those options which seem to have been of interest - please shout if some are missing in the polls.

**Comment by ericpre** [comment]
🚀 for Noto Sans Display Semi-bold
🎉 for Radio Canda Semi-bold
👀 for Lato

**Comment by ericpre** [comment]
From the polls, we can narrow down to 2 candidates:
- Lato with different `IO` from Noto Sans Display Semi-bold
- Noto Sans Display Semi-bold

@jordiferrero: could please make these two (final?) candidates with the snake?

**Comment by jordiferrero** [comment]
> From the polls, we can narrow down to 2 candidates:
> 
> * Lato with different `IO` from Noto Sans Display Semi-bold
> * Noto Sans Display Semi-bold
> 
> @jordiferrero: could please make these two (final?) candidates with the snake?

OK. Here are the final versions: let's make a final final poll. 🎉 --> Lato + Noto   OR   🚀--> Noto

Lato + Noto 🎉:
![lato_noto](https://user-images.githubusercontent.com/9082687/194722585-3276fe0f-3c69-4fbb-820e-875860acf2ca.png)

Noto 🚀:
![noto](https://user-images.githubusercontent.com/9082687/194722584-33f8c65f-ecf3-4345-8d44-871aca29cfaf.png)

**Comment by ericpre** [comment]
It sounds like that RosettaSciIO now has a logo! :)
@jordiferrero, do you want to make a pull request with the exported file and raw file (for potential future change)? Thank you!
